### PR TITLE
Update Subfields.yaml

### PR DESCRIPTION
**Pull request by JennyChan-ptpl**
Additional four new subfield; ballast, lighting, auto, manual

### CODE REVIEW

**Comment by charbull** [comment]
@JennyChan-ptpl can you please sign the CLA?

**Comment by JennyChan-ptpl** [comment]
> @JennyChan-ptpl can you please sign the CLA?

@googlebot I signed it!

**Comment by charbull** [comment]
Thank you @JennyChan-ptpl 
waiting for reviews before merging

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (tasodorff) [inline ontology/yaml/resources/subfields/subfields.yaml:21]: Proposed, more informative definition: "An electrical ballast is a device placed in series with a load to limit the amount of current in an electrical circuit. Commonly a component of light fixtures."

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (tasodorff) [inline ontology/yaml/resources/subfields/subfields.yaml:40]: this should be adjusted to `light` or similar if its going to be a component (i.e. a discrete, countable, instance of a thing);

if the intention is for this to be a descriptor, `lighting` is acceptable; it would need to be moved however.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (tasodorff) [inline ontology/yaml/resources/subfields/subfields.yaml:21]: Proposed, more informative definition: "An electrical ballast is a device placed in series with a load to limit the amount of current in an electrical circuit. Commonly a component of light fixtures."

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml:40]: I agree with you to change to "light" or "lamp" so that more lighting device can use this component subfield. The widely usage of "light" or "lamp" component is greater then "lighting" as a descriptor.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: @tasodorff description had been updated. Appreciate your suggestion.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (pisuke) [inline ontology/yaml/resources/subfields/subfields.yaml:21]: The term "ballast" typically refers only to fluorescent lighting, while the term driver is used for LEDs.
Here is some [reference](http://www.needhamelectric.com/ignites-lights-ballasts-vs-drivers/#:~:text=Fluorescent%20ballasts%20provide%20an%20initial,anode%20within%20the%20discharge%20tube.&text=LED%20drivers%20convert%20high%20voltage,are%20designed%20to%20run%20on.) for context.
A more generic term could be "control gear".
We should add both "ballast" and "driver".

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (pisuke) [inline ontology/yaml/resources/subfields/subfields.yaml:40]: "Light" is not a thing or component, but a phenomenon. I think it is too generic to be included for a specific component.
It's best to use "lamp" for a light source and "daylight_device" or something more specific like "window", "reflector", etc. for a daylight component.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (pisuke) [inline ontology/yaml/resources/subfields/subfields.yaml:50]: perhaps better to use "automatic"?

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (pisuke) [reply ontology/yaml/resources/subfields/subfields.yaml]: I've recommended to use "lamp" as a source of light, because light is a phenomenon, not a component.
See my review comments.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (pisuke) [reply ontology/yaml/resources/subfields/subfields.yaml]: I've recommend to add also "driver" because "ballast" refers to fluorescent lighting only. See my other comments in the review.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml:21]: @pisuke thanks for the comment. "driver" is added

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: @pisuke "lamp" is used as artificial light source devices, and description of daylight had been removed.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: Yes I agree with you

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: "driver" is added.

File: ontology/yaml/resources/subfields/subfields.yaml
Comment (tasodorff) [inline ontology/yaml/resources/subfields/subfields.yaml:33]: Is this different from controller above? If not, we should choose one to use and remove the other.

Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml:33]: Yes it is different from controller. @pisuke recommended to add "driver" for LED to differentiate with "ballast" which is only used for fluorescent light.

Comment (charbull) [reply ontology/yaml/resources/subfields/subfields.yaml:33]: @tasodorff @pisuke any comments?

Comment (tasodorff) [reply ontology/yaml/resources/subfields/subfields.yaml:33]: this sounds like `controller` to me, just specific to the lighting domain. `driver` has multiple meanings outside this (e.g. a network driver). 

if you were to distinguish it from a controller, how would you?

Comment (charbull) [reply ontology/yaml/resources/subfields/subfields.yaml:33]: @JennyChan-ptpl friendly ping :)

Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml:33]: @charbull A motor controllers provide feedback, error detection, and other features which usually aren't found on basic motor driver boards.

Comment (tasodorff) [reply ontology/yaml/resources/subfields/subfields.yaml:33]: So then we should adjust the driver definition to explicitly state that it provides the motor (or lighting board) with power, but doesn't handle logical control of the device.

Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: Thanks for your suggestion, the driver definition had been updated.
driver: "A circuit or component that provides the motor (or lighting board) with power, but doesn't handle logical control of the device."

Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: Any further updates needed?

Comment (JennyChan-ptpl) [reply ontology/yaml/resources/subfields/subfields.yaml]: @tasodorff , If you have no further comments, can you help to merge to the master copy?

**Comment by charbull** [comment]
@tasodorff @pisuke are merging this one? or closing it?

**Review by tasodorff — APPROVED** [review]

**Comment by tasodorff** [comment]
@pisuke please review and see if changes are OK

**Comment by tasodorff** [comment]
closed due to inactivity.

**Pull request closed without merging**
